src/main/scala/io/iohk/ethereum/ledger/InMemoryWorldStateProxy.scala

```diff
@@ -155,7 +155,8 @@ class InMemoryWorldStateProxy private(
     copyWith(accountsStateTrie = accountsStateTrie.put(address.bytes, account))
 
   override def deleteAccount(address: Address): InMemoryWorldStateProxy =
-    copyWith(accountsStateTrie = accountsStateTrie.remove(address.bytes))
+    copyWith(accountsStateTrie = accountsStateTrie.remove(address.bytes),
+      contractStorages = contractStorages - address)
```

Comment:
`accountCodes` must be updated elsewhere, as my tests work without changing it. Or maybe there is another bug, that we should solve so that we have a bug with `accountCodes` that we can solve.